packages/ui/src/models/camel/camel-route-resource.ts

```diff
@@ -46,12 +48,14 @@ export class CamelRouteResource implements CamelResource, BeansAwareResource {
   ) => number;
   private entities: BaseCamelEntity[] = [];
   private resolvedEntities: BaseVisualCamelEntityDefinition | undefined;
-  private comments: string[] = [];
+  private serializer: CamelResourceSerializer;
 
-  constructor(json?: unknown) {
-    if (!json) return;
-    const rawEntities = Array.isArray(json) ? json : [json];
-    this.entities = rawEntities.reduce((acc, rawItem) => {
+  constructor(code?: unknown, serializer?: CamelResourceSerializer) {
```

Comment:
Let's try to use the `CamelYamlDsl` type

```suggestion
  constructor(code?: CamelYamlDsl, serializer?: CamelResourceSerializer) {
```

Reply:
Please try to find a better name than the existing `code` :pray: :package: